fbw-a380x/src/wasm/systems/a380_systems/src/hydraulic/mod.rs

```diff
@@ -5663,6 +5660,113 @@ impl SimulationElement for ElevatorSystemHydraulicController {
 }
 
+#[derive(Clone, Copy, PartialEq)]
+struct ThsMotorElectricalController {
+    should_motor_activate: bool,
+    requested_position: Angle,
+}
+impl ThsMotorElectricalController {
+    fn new() -> Self {
+        Self {
+            should_motor_activate: false,
+
+            requested_position: Angle::default(),
+        }
+    }
+
+    fn set_mode(&mut self, should_motor_activate: bool) {
+        self.should_motor_activate = should_motor_activate;
+    }
+
+    /// Receives a [0;1] position request, 0 is down 1 is up
+    fn set_requested_position(&mut self, requested_position: Angle) {
+        self.requested_position = requested_position;
+    }
+}
+impl ThsMotorController for ThsMotorElectricalController {
+    fn motor_should_activate(&self) -> bool {
+        self.should_motor_activate
+    }
+
+    fn requested_position(&self) -> Angle {
+        self.requested_position
+    }
+}
+
+struct ThsSystemHydraulicController {
+    ths_green_actuator_solenoid_id: VariableIdentifier,
+    ths_yellow_actuator_solenoid_id: VariableIdentifier,
+
+    ths_green_actuator_position_demand_id: VariableIdentifier,
+    ths_yellow_actuator_position_demand_id: VariableIdentifier,
+
+    position_requests_from_fbw: [Angle; 2],
+    solenoid_energized_from_fbw: [bool; 2],
+
+    controllers: [ThsMotorElectricalController; 2],
+}
+impl ThsSystemHydraulicController {
+    fn new(context: &mut InitContext) -> Self {
+        Self {
+            ths_green_actuator_solenoid_id: context
+                .get_identifier("THS_GREEN_SERVO_SOLENOID_ENERGIZED".to_owned()),
+            ths_yellow_actuator_solenoid_id: context
+                .get_identifier("THS_YELLOW_SERVO_SOLENOID_ENERGIZED".to_owned()),
+
+            ths_green_actuator_position_demand_id: context
+                .get_identifier("THS_GREEN_COMMANDED_POSITION".to_owned()),
+            ths_yellow_actuator_position_demand_id: context
+                .get_identifier("THS_YELLOW_COMMANDED_POSITION".to_owned()),
+
+            position_requests_from_fbw: [Angle::default(); 2],
+            solenoid_energized_from_fbw: [false; 2],
+
+            // Controllers are in green->yellow order
+            controllers: [
+                ThsMotorElectricalController::new(),
+                ThsMotorElectricalController::new(),
+            ],
+        }
+    }
+
+    fn controllers(&self) -> &[ThsMotorElectricalController; 2] {
+        &self.controllers
+    }
+
+    fn update(&mut self) {
+        self.update_ths_controllers_positions();
+        self.update_ths_controllers_solenoids();
+    }
+
+    fn update_ths_controllers_positions(&mut self) {
+        self.controllers[ThsMotorPosition::Green as usize].set_requested_position(
+            self.position_requests_from_fbw[ThsMotorPosition::Green as usize],
+        );
+        self.controllers[ThsMotorPosition::Yellow as usize].set_requested_position(
+            self.position_requests_from_fbw[ThsMotorPosition::Yellow as usize],
+        );
```

Comment:
```suggestion
        for (controller, position_request) in self
            .controllers
            .iter_mut()
            .zip(self.position_requests_from_fbw)
        {
            controller.set_requested_position(position_request);
        }
```

Reply:
I think in these cases I'd rather explicitly go through the array, asthere are only two controllers, the looped variant is longer, and also IIRC in the rest of the file, the position requests are all explicitly listed too